pandas/core/groupby/groupby.py

```diff
@@ -663,7 +665,17 @@ def get_converter(s):
             converter = get_converter(index_sample)
             names = (converter(name) for name in names)
 
-        return [self.indices.get(name, []) for name in names]
+        # return [self.indices.get(name, []) for name in names]
+        # self.indices is a dict and doesn't handle looking up nulls in the groups
+        from pandas import (
+            Index,
+            Series,
+        )
+
+        index = Index(self.indices.keys(), tupleize_cols=False)
+        indices = Series(self.indices.values(), index=index)
+        result = [indices[name] for name in names]
+        return result
```

Comment:
Here `self.indices` is a dict mapping the name to the indices it occurs at. Problem is when there are null values for the name (e.g. `np.nan`, `(1, np.nan)`, etc). Here the lookup fails, and in general one shouldn't have null values as keys in a dict. I know Series gets around this using codes, hence the solution above.

The need to separate out the Index carefully is to handle odd cases like where `self.indices` is `{(1,): np.array([0, 2]), (1, 2): np.array([1, 3])}`. This is the particular case for a single test.

cc @jbrockmendel @phofl if there are any ideas to solve this in some more direct way rather than using a Series.

Reply:
> Here the lookup fails, and in general one shouldn't have null values as keys in a dict

Not sure if this is relevant, but possibly related to #43943?

Reply:
i think in index.pyx we have some code for canonicalizing NA values, e.g. float("nan") -> np.nan, so that dict lookups are better-behaved.  could be adapted/reused when defining `indices`?

Reply:
Thanks - this is precisely what I was looking for. However, I found a simpler way to use the codes directly; I've put up #45953.